src/decorators.ts

```diff
@@ -0,0 +1,192 @@
+            };
+
+        target.getProperties = // Legacy
+            target.amorphicGetProperties = function getProperties(includeVirtual) {
+                return objectTemplate._getDefineProperties(target, undefined, includeVirtual);
+            };
+
+        target.createProperty = // Legacy
+            target.amorphicCreateProperty = function (propertyName, defineProperty) {
+                if (defineProperty.body) {
+                    target.prototype[propertyName] = objectTemplate._setupFunction(propertyName, defineProperty.body,
+                        defineProperty.on, defineProperty.validate);
+                }
+                else {
+                    target.prototype.__amorphicprops__[propertyName] = defineProperty;
+                    if (typeof defineProperty.value in ['string', 'number'] || defineProperty.value == null) {
+                        Object.defineProperty(target.prototype, propertyName,
+                            { enumerable: true, writable: true, value: defineProperty.value });
+                    }
+                    else {
+                        Object.defineProperty(target.prototype, propertyName, {
+                            enumerable: true,
+                            get: function () {
+                                if (!this['__' + propertyName]) {
+                                    this['__' + propertyName] =
+                                        ObjectTemplate.clone(defineProperty.value, defineProperty.of ||
+                                            defineProperty.type || null);
+                                }
+                                return this['__' + propertyName];
+                            },
+                            set: function (value) {
+                                this['__' + propertyName] = value;
+                            }
+                        });
+                    }
+                }
+            };
+
+        if (target.prototype.__exceptions__) {
+            objectTemplate.__exceptions__ = objectTemplate.__exceptions__ || [];
+            for (var exceptionKey in target.prototype.__exceptions__) {
+                objectTemplate.__exceptions__.push({
+                    func: target.prototype.__exceptions__[exceptionKey],
+                    class: getName,
+                    prop: exceptionKey
+                });
+            }
+        }
+
+        function defineProperties() {
+            return target.prototype.__amorphicprops__;
+        }
+        function getName() {
+            return target.toString().match(/function ([^(]*)/)[1];
+        }
+        function getDictionary() {
+            objectTemplate.getClasses();
+        }
+        function getParent() {
+            getDictionary();
+            return target.__shadowParent__;
+        }
+        function getChildren() {
+            getDictionary();
+            return target.__shadowChildren__;
+        }
+
+        /*
+        TODO: Typescript
```

Comment:
we should track this TODO in ticket

Reply:
Right, I'm not too sure what this TODO is for anymore. it's been too long.